Function `disable_nagle` moved to Sockets and documented

mateuszbaran opened the pull request:
I've moved the `diable_nagle` functions from Distributed to Sockets, as suggested in #31842 . I was trying to make the PR small, so:
1. `disable_nagle` precompile statements should probably go to the Sockets stdlib.
2. I was thinking about adding some tests but libuv doesn't provide `getsockopt` to test that an option is actually set.

jpsamaroo commented:
Nitpick: Spelling of "disable_nagle" in commit message

Also, probably should have a NEWS entry?

StefanKarpinski commented:
How about changing the API to `nagle(::Bool)` instead whole we’re at it?

mateuszbaran commented:
@jpsamaroo thanks, I've fixed the commit message (and the PR title). I haven't exported `disable_nagle` so it probably doesn't need a NEWS entry.

If we are going to change the API, then maybe we should stay closer to the naming from libuv? For example `tcp_nodelay(socket::Union{TCPServer, TCPSocket}, enable::Bool)`.
BTW, I've researched the `TCP_QUICKACK` flag that is also set in `disable_nagle` and it doesn't seem appropriate to bundle these two in a single function. `TCP_NODELAY` is set permanently for a given socket while `TCP_QUICKACK` is apparently reset by certain operations ( https://github.com/nodejs/node/issues/21104#issuecomment-435102382 ).

JeffBezanson commented:
Ok, two functions sounds fine. `tcp_nodelay` and `tcp_quickack`? (Although I hate the double negative in `nodelay=false`...)

mateuszbaran commented:
OK, if you don't like `tcp_nodelay` then I'll name it `nagle` (and the other one will be `tcp_quickack`).

mateuszbaran committed:
disable_nagle moved to Sockets, split and documented (see #31842 )

disable_nagle was split into nagle (which enables or disables Nagle's 
algorithm) and quickack (which enables or disables TCP_QUICKACK on Linux 
systems).

mateuszbaran commented:
I've split `disable_nagle` to `nagle` and `quickack` and added `enable` arguments. Is this OK?

StefanKarpinski commented:
I'm pretty happy with this now but it does make me wonder if we shouldn't expose these as properties of the socket, i.e. `sock.nagle` and `sock.quickack` which can be read or assigned. @vtjnash or @keno, do you have strong feelings about that?

vtjnash commented on stdlib/Sockets/src/Sockets.jl:
```diff
@@ -471,6 +471,31 @@ function connect(sock::LibuvStream, args...)
+"""
+function nagle(sock::Union{TCPServer, TCPSocket}, enable::Bool)
+    # disable or enable Nagle's algorithm on all OSes
+    ccall(:uv_tcp_nodelay, Cint, (Ptr{Cvoid}, Cint), sock.handle, Cint(!enable))
```
Error checking of return value?

mateuszbaran replied:
Good idea, I'll make a separate PR for this.
